src/bun.js/webcore/streams.zig

```diff
@@ -3029,7 +3037,7 @@ pub const FileSink = struct {
         // TODO: this should be concurrent.
         const fd = switch (switch (options.input_path) {
             .path => |path| bun.sys.openA(path.slice(), options.flags(), options.mode),
-            .fd => |fd_| bun.sys.dupWithFlags(fd_, if (bun.FDTag.get(fd_) == .none) std.os.O.NONBLOCK else 0),
+            .fd => |fd_| bun.sys.dupWithFlags(fd_, if (bun.FDTag.get(fd_) == .none and !this.force_sync_on_windows) std.os.O.NONBLOCK else 0),
```

Comment:
Any chance with the `!this.force_sync_on_windows` on 3040 and 3063 that it's the cheaper check?
If so moving it to the front of the `and` chain is the recommendation.